assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js

```diff
@@ -71,24 +66,15 @@ function AudienceSegmentationSetupCTAWidget( { Widget } ) {
 		select( MODULES_ANALYTICS_4 ).getConfiguredAudiences()
 	);
 
-	const hasDataWithinPast90Days = useSelect( ( select ) => {
-		const endDate = select( CORE_USER ).getReferenceDate();
-
-		const startDate = getPreviousDate(
-			endDate,
-			90 + DATE_RANGE_OFFSET // Add offset to ensure we have data for the entirety of the last 90 days.
-		);
-
-		const args = {
-			metrics: [ { name: 'totalUsers' } ],
-			startDate,
-			endDate,
-		};
-
-		return select( MODULES_ANALYTICS_4 ).hasZeroData( args ) === false;
+	const analyticsIsDataAvailableOnLoad = useSelect( ( select ) => {
+		select( MODULES_ANALYTICS_4 ).isGatheringData();
+		return select( MODULES_ANALYTICS_4 ).isDataAvailableOnLoad();
 	} );
 
-	if ( configuredAudiences !== null || ! hasDataWithinPast90Days ) {
+	if (
+		( configuredAudiences && configuredAudiences.length ) ||
+		! analyticsIsDataAvailableOnLoad
+	) {
```

Comment:
This doesn't check for `undefined` but only for the truthy value, and the array is not empty.

Reply:
@hussain-t This is checking if `configuredAudiences` are present. If yes, then we do not need to render the CTA widget as we already have setup the audiences.

Thoughts?

Reply:
Makes sense 👍